akka-persistence-query/src/main/scala/akka/persistence/query/PersistenceQuery.scala

```diff
@@ -24,23 +21,19 @@ object PersistenceQuery extends ExtensionId[PersistenceQuery] with ExtensionIdPr
 
   def createExtension(system: ExtendedActorSystem): PersistenceQuery = new PersistenceQuery(system)
 
-  def lookup() = PersistenceQuery
+  def lookup(): PersistenceQuery.type = PersistenceQuery
 
-  /** INTERNAL API. */
-  private[persistence] case class PluginHolder(
-    scaladslPlugin: scaladsl.ReadJournal, javadslPlugin: akka.persistence.query.javadsl.ReadJournal)
-    extends Extension
+  @InternalApi
+  private[akka] implicit val pluginProvider = new PluginProvider[ReadJournalProvider, scaladsl.ReadJournal, javadsl.ReadJournal] {
+    override def scalaDsl(t: ReadJournalProvider): ReadJournal = t.scaladslReadJournal()
+    override def javaDsl(t: ReadJournalProvider): javadsl.ReadJournal = t.javadslReadJournal()
+  }
 
 }
 
-class PersistenceQuery(system: ExtendedActorSystem) extends Extension {
-  import PersistenceQuery._
-
-  private val log = Logging(system, getClass)
-
-  /** Discovered query plugins. */
-  private val readJournalPluginExtensionIds = new AtomicReference[Map[String, ExtensionId[PluginHolder]]](Map.empty)
+import PersistenceQuery.pluginProvider
```

Comment:
add comment that this is about the implicit

Reply:
could we move it to some more specific place than at the top file level?

Reply:
I'll change this to pass in the implicit explicitly so nothing needs to be implicit